lib/krane/template_sets.rb

```diff
@@ -39,21 +40,31 @@ def ejson_secrets_file
 
       def validate
         errors = []
+        supported_extensions = if @render_erb
+          TemplateSets::VALID_TEMPLATES
+        else
+          TemplateSets::VALID_TEMPLATES.reject { |extension| extension.include?('erb') }
+        end
+
         if Dir.entries(@template_dir).none? do |filename|
-             filename.end_with?(*TemplateSets::VALID_TEMPLATES) ||
+             filename.end_with?(*supported_extensions) ||
              filename.end_with?(EjsonSecretProvisioner::EJSON_SECRETS_FILE)
            end
-          return errors << "Template directory #{@template_dir} does not contain any valid templates"
+          return errors << "Template directory #{@template_dir} does not contain any valid templates (supported " \
+              "suffixes: #{supported_extensions.join(', ')}, or #{EjsonSecretProvisioner::EJSON_SECRETS_FILE})"
         end
 
         @files.each do |filename|
           filename = File.join(@template_dir, filename)
           if !File.exist?(filename)
             errors << "File #{filename} does not exist"
-          elsif !filename.end_with?(*TemplateSets::VALID_TEMPLATES) &&
+          elsif filename.end_with?("erb") && !@render_erb
+            errors << "Rendering erb during deploy is not supported. Try piping the output of `krane render` "\
```

Comment:
Can we ensure we only emit the error once? I'm also worried this error is a bit to specific, e.g. if its triggered from a different situation it would be rather confusing. Maybe something slightly more generic: 
"ERB template discovered with rendering disabled. If you were trying to deploy try piping..."

![Screen Shot 2019-11-17 at 9 30 48 AM](https://user-images.githubusercontent.com/12193286/69011205-04f3a780-091d-11ea-9d63-485ea51c679e.png)

(^ doesn't happen right now since erb is enabled from deploy, but it would with 632)

Reply:
Yep on it now.